osu.Game/Screens/Play/Player.cs

```diff
@@ -154,6 +159,10 @@ protected override void LoadComplete()
             // replays should never be recorded or played back when autoplay is enabled
             if (!Mods.Value.Any(m => m is ModAutoplay))
                 PrepareReplay();
+
+            // needs to be bound here as the last binding, otherwise cases like starting a replay while not focused causes player to exit, if activity is bound before checks.
```

Comment:
Can you explain? I would like to think the `HasReplayLoaded` check in the callback should suffice.

Reply:
if game activity is bound before the guards are and game is inactive, then calling the method before all guards are set to correct values may result in incorrect actions performed.

but actually now that you said it, I think it could be removed as I've corrected `IsBreakTime` to stay `true` until it's cleared on first `Update()`.

Reply:
I don't think `IsBreakTime` should be `true` by default.

And you haven't actually answered my question, so I'll answer it myself: the `PrepareReplay()` call *sets* `HasReplayLoaded`. I'd maybe reword the comment to mention that more directly.

Reply:
ah, I haven't actually thought about the `PrepareReplay()` call above, just put this in a place where it should be the last thing to be called, will reword then.

Reply:
hmm, since I've changed `performUserRequestedExit` to calling `Pause()` directly, the comment looks to be completely redundant, I'm a bit concerned about whether to remove it as that sounds like it would hide a potential issue.. what do you think?

Reply:
If the order of operations no longer matters then I'd just remove the comment.